azure_functions_worker/extension.py

```diff
@@ -0,0 +1,297 @@
+                    elif cls._is_post_invocation_hook(hook_name):
+                        hook_meta.ext_impl(ext_logger, ctx, fargs, fret)
+                except Exception as e:
+                    ext_logger.error(e, exc_info=True)
+
+    @classmethod
+    def _safe_execute_function_load_hooks(cls, hooks, hook_name, fname, fdir):
+        # hooks from azure.functions.ExtensionMeta.get_function_hooks() or
+        #            azure.functions.ExtensionMeta.get_application_hooks()
+        if hooks:
+            # Invoke extension implementation from .<hook_name>.ext_impl
+            for hook_meta in getattr(hooks, hook_name, []):
+                try:
+                    hook_meta.ext_impl(fname, fdir)
+                except Exception as e:
+                    logger.error(e, exc_info=True)
+
+    @classmethod
+    def _raw_invocation_wrapper(cls, ctx, function, args) -> Any:
+        """Calls pre_invocation and post_invocation extensions additional
+        to function invocation
+        """
+        cls._invocation_extension(ctx, APP_EXT_PRE_INVOCATION, args)
+        cls._invocation_extension(ctx, FUNC_EXT_PRE_INVOCATION, args)
+        result = function(**args)
+        cls._invocation_extension(ctx, FUNC_EXT_POST_INVOCATION, args, result)
+        cls._invocation_extension(ctx, APP_EXT_POST_INVOCATION, args, result)
+        return result
+
+    @classmethod
+    def _try_get_sdk_with_extension_enabled(cls) -> Optional[ModuleType]:
+        if cls._is_sdk_detected:
+            return cls._extension_enabled_sdk
+
+        sdk = cls._get_sdk_from_sys_path()
+        if cls._is_extension_enabled_in_sdk(sdk):
+            cls._info_extension_is_enabled(sdk)
+            cls._extension_enabled_sdk = sdk
+        else:
+            cls._warn_sdk_not_support_extension(sdk)
+            cls._extension_enabled_sdk = None
+
+        cls._is_sdk_detected = True
+        return cls._extension_enabled_sdk
+
+    @classmethod
+    def _info_extension_is_enabled(cls, sdk):
+        logger.info(
+            'Python Worker Extension is enabled in azure.functions '
+            f'({cls._get_sdk_version(sdk)}).'
+        )
+
+    @classmethod
+    def _info_discover_extension_list(cls, function_name, sdk):
+        logger.info(
+            f'Python Worker Extension Manager is loading {function_name}, '
+            'current registered extensions: '
+            f'{sdk.ExtensionMeta.get_registered_extensions_json()}'
+        )
+
+    @classmethod
+    def _warn_sdk_not_support_extension(cls, sdk):
+        logger.warning(
+            f'The azure.functions ({cls._get_sdk_version(sdk)}) does not '
+            'support Python worker extensions. If you believe extensions '
+            'are correctly installed, please set the '
+            f'{PYTHON_ISOLATE_WORKER_DEPENDENCIES} and '
+            f'{PYTHON_ENABLE_WORKER_EXTENSIONS} to "true"'
+        )
+
+    @classmethod
+    def _warn_context_has_no_function_name(cls):
```

Comment:
I didn't see any usage of this method. Can you double check?

Reply:
Good catch, I will remove this.